controllers/host/profile_utils.go

```diff
@@ -572,3 +575,64 @@ func (r *HostReconciler) ValidateProfile(host *starlingxv1.Host, profile *starli
 
 	return nil
 }
+
+// Sync interface name for ethernet
+// When N3000 interface is used, it is possible to change name after unlocked.
+// This function is to copy interface name from current to profile if uuid is the same
+// to avoid configuration difference because of names.
+func SyncIFNameByUuid(
+	profile *starlingxv1.HostProfileSpec, current *starlingxv1.HostProfileSpec,
+) {
+	if profile == nil || current == nil {
+		return
+	}
+
+	if_current := current.Interfaces.Ethernet
+	if_profile := profile.Interfaces.Ethernet
+	for idx_current := range if_current {
+		info_current := if_current[idx_current].CommonInterfaceInfo
+		port_current := if_current[idx_current].Port.Name
+		for idx_profile := range if_profile {
+			info_profile := if_profile[idx_profile].CommonInterfaceInfo
+			if info_profile.UUID == info_current.UUID &&
+				info_profile.Name != info_current.Name {
+				logProfileUtils.Info(
+					"Ethernet name sync", "profile", info_profile.Name,
+					"current", info_current.Name, "uuid", info_profile.UUID)
+				if_profile[idx_profile].CommonInterfaceInfo.Name = info_current.Name
+				if_profile[idx_profile].Port.Name = port_current
+				break
+			}
+		}
+	}
+}
+
+// FillEmptyUuidbyName fills the empty uuid by its name. Normally, the uuids can
+// be searched from the system and merged to the default profile.
+func FillEmptyUuidbyName(
+	profile *starlingxv1.HostProfileSpec, current *starlingxv1.HostProfileSpec,
+) {
+	if profile == nil || current == nil {
+		return
+	}
+
+	if_current := current.Interfaces.Ethernet
+	if_profile := profile.Interfaces.Ethernet
+	for idx_profile := range if_profile {
+		info_profile := if_profile[idx_profile].CommonInterfaceInfo
+		profile_uuid := info_profile.UUID
+		if profile_uuid == "" || profile == nil {
```

Comment:
I think `|| profile == nil` should be removed from here.
If `profile == nil` it would return at line 616 and doesn't come this far though.

Reply:
Acked

Reply:
I think that's a mistake that I need to check if the "profile_uuid" is nil